Retry if a display is not found

I use display-switch to workaround an issue where MacOS sometimes does not wake-up its external display when switching between a Macbook and Linux desktop via a KVM.

```
Macbook (closed lid) -
                       \
                         -->  KVM  -->  External display on DisplayPort1
                       /
Linux desktop --------
```

I have configured display-switch on my Macbook to monitor a USB device that is connected to the KVM and to send a DDC input-switch command to my monitor whenever the USB device is connected. Although, the monitor is always connected to DisplayPort1 (no other video cables are connected), the input-switch command (or perhaps a wake-up command is also sent?) has the side-effect of waking the monitor up, which solves the wake-up flakiness issue I was experiencing.

This usually works:

```
19:27:26 [INFO] Monitored device ("14b0:011e") is connected
19:27:28 [DEBUG] (1) display_switch::display_control: Setting display 'DELL U3417W' to DisplayPort1(0xf)
```

But sometimes, due I think to timing of the USB detection and KVM port switching, the monitor is not detected:
```
15:00:26 [INFO] Monitored device ("14b0:011e") is connected
15:00:27 [ERROR] Did not detect any DDC-compatible displays!
```

I believe that this issue could be worked around by introducing a retry mechanism. Perhaps something with a slight delay - even a fixed 2 second sleep before retrying would probably be sufficient.

--- a/src/display_control.rs
+++ b/src/display_control.rs
@@ -7,6 +7,7 @@ use crate::configuration::{Configuration, SwitchDirection};
 use crate::input_source::InputSource;
 use ddc_hi::{Ddc, Display};
 use std::collections::HashSet;
+use std::{thread, time};
 
 /// VCP feature code for input select
 const INPUT_SELECT: u8 = 0x60;
@@ -24,8 +25,19 @@ fn are_display_names_unique(displays: &[Display]) -> bool {
     displays.iter().all(|display| hash.insert(display_name(display, None)))
 }
 
+fn displays() -> Vec<Display> {
+    let mut displays = Display::enumerate();
+    if displays.is_empty() {
+        let retry_duration = time::Duration::from_millis(2000);
+        warn!("Did not detect any DDC-compatible displays. Retrying after {} second(s)...", retry_duration.as_secs());
+        thread::sleep(retry_duration);
+        displays = Display::enumerate();
+    }
+    return displays;
+}
+
 pub fn log_current_source() {
-    let displays = Display::enumerate();
+    let displays = displays();
     if displays.is_empty() {
         error!("Did not detect any DDC-compatible displays!");
         return;
@@ -46,7 +58,7 @@ pub fn log_current_source() {
 }
 
 pub fn switch(config: &Configuration, switch_direction: SwitchDirection) {
-    let displays = Display::enumerate();
+    let displays = displays();
     if displays.is_empty() {
         error!("Did not detect any DDC-compatible displays!");
         return;

--- a/src/display_control.rs
+++ b/src/display_control.rs
@@ -26,14 +26,18 @@ fn are_display_names_unique(displays: &[Display]) -> bool {
 }
 
 fn displays() -> Vec<Display> {
-    let mut displays = Display::enumerate();
-    if displays.is_empty() {
-        let retry_duration = time::Duration::from_millis(2000);
-        warn!("Did not detect any DDC-compatible displays. Retrying after {} second(s)...", retry_duration.as_secs());
-        thread::sleep(retry_duration);
-        displays = Display::enumerate();
+    let displays = Display::enumerate();
+    if !displays.is_empty() {
+        return displays
     }
-    return displays;
+
+    // Under some conditions, such as when using a KVM, it's possible for the USB connection/disconnection events to
+    // occur before the display(s) become available. We retry once after a bit of a delay in order to be more
+    // forgiving with regard to timing.
+    let retry_duration = time::Duration::from_millis(2000);
+    warn!("Did not detect any DDC-compatible displays. Retrying after {} second(s)...", retry_duration.as_secs());
+    thread::sleep(retry_duration);
+    return Display::enumerate();
 }
 
 pub fn log_current_source() {

--- a/src/display_control.rs
+++ b/src/display_control.rs
@@ -11,6 +11,7 @@ use std::{thread, time};
 
 /// VCP feature code for input select
 const INPUT_SELECT: u8 = 0x60;
+const RETRY_DELAY_MS: u64 = 2000;
 
 fn display_name(display: &Display, index: Option<usize>) -> String {
     if let Some(index) = index {
@@ -34,9 +35,9 @@ fn displays() -> Vec<Display> {
     // Under some conditions, such as when using a KVM, it's possible for the USB connection/disconnection events to
     // occur before the display(s) become available. We retry once after a bit of a delay in order to be more
     // forgiving with regard to timing.
-    let retry_duration = time::Duration::from_millis(2000);
-    warn!("Did not detect any DDC-compatible displays. Retrying after {} second(s)...", retry_duration.as_secs());
-    thread::sleep(retry_duration);
+    let delay_duration = time::Duration::from_millis(RETRY_DELAY_MS);
+    warn!("Did not detect any DDC-compatible displays. Retrying after {} second(s)...", delay_duration.as_secs());
+    thread::sleep(delay_duration);
     return Display::enumerate();
 }
 

